src/DependencyInjection/CsaGuzzleExtension.php

```diff
@@ -95,18 +96,12 @@ private function processCacheConfiguration(array $config, ContainerBuilder $cont
     private function processClientsConfiguration(array $config, ContainerBuilder $container, Definition $clientFactory, Definition $descriptionFactory)
     {
         foreach ($config['clients'] as $name => $options) {
-            $clientFactory->addMethodCall('registerClientConfiguration', [
-                $name,
-                $options['config'],
-                $options['subscribers']
+            $client = new Definition($config['factory_class']);
+            $client->addArgument($options['config']);
+            $client->addTag(SubscriberPass::CLIENT_TAG, [
+                'subscribers' => implode(', ', $options['subscribers'])
```

Comment:
The `$options['subscribers']` array actually hold the following values:

```php
$options['subscribers'] = [
    'logger' => true,
    'cache' => true,
];
```

or 

```php
$options['subscribers'] = [
    'cache' => false,
];
```

This is done in order to disable some subscribers, if the value is set to `false`. Currently, all subscribers are registered, except for the ones which are false.

See https://github.com/csarrazi/CsaGuzzleBundle/blob/master/src/Factory/ClientFactory.php#L69

Here, we fetch the complete list of subscriber names, and filter subscribers for which the configuration is set to false.

This can change for release `1.3.0`, but it should be addressed in another issue as, if this is changed, we need to ensure compatibility with the old system (and raise a deprecation notice).

For now, you can import the current logic from the factory's `createNamed()` method.